### PR TITLE
Feature/contributor metadata

### PR DESCRIPTION
**Pull request by jacobwindsor**
For [#204](https://app.zenhub.com/workspaces/dcp-ingest-product-development-5f71ca62a3cb47326bdc1b5c/issues/ebi-ait/dcp-ingest-central/204)

### CODE REVIEW

**Comment by jacobwindsor** [comment]
[Integration tests passing](https://gitlab.ebi.ac.uk/hca/ingest-integration-tests/-/pipelines/139538)

File: client/src/app/project-registration/project-registration-form/project-reg-layout.ts [hunk 1]
```diff
@@ -22,53 +31,52 @@ export const projectRegLayout: MetadataFormLayout = {
...
     },
     {
-      title: 'Summary',
-      key: 'summary',
+      title: 'Funders',
+      key: 'project.content.funders',
+      items: [
+        {
+          keys: [
+            'project.content.contributors'
```
Comment (amnonkhen) [inline client/src/app/project-registration/project-registration-form/project-reg-layout.ts:67]: should this be `funders`?

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: client/src/app/project-registration/project-registration-form/project-registration-form.component.html [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
-                     [config]="config"
-                     [selectedTabKey]="formTabKey"
-                     (cancel)="onCancel($event)"
-                     (back)="decrementTab()"
-                     (tabChange)="onTabChange($event)"
-                     (save)="onSave($event)">
-  </app-metadata-form>
-</ng-container>
+  <ng-container *ngIf="userIsWrangler">
+    <mat-tab label="3. Data upload" disabled></mat-tab>
```
Comment (amnonkhen) [inline client/src/app/project-registration/project-registration-form/project-registration-form.component.html:25]: *3. Data Upload* can be moved out of the conditional, can't it?

File: client/src/app/project/project.component.ts [hunk 3]
```diff
@@ -38,19 +38,51 @@ export class ProjectComponent implements OnInit {
   processesDataSource: MetadataDataSource<MetadataDocument>;
   filesDataSource: MetadataDataSource<MetadataDocument>;
 
+  sharedTabConfig = [
+    {
+      key: 'project',
+      label: '1. Project'
```
Comment (amnonkhen) [inline client/src/app/project/project.component.ts:44]: I think these texts are duplicated in `project-registration-form.component.html`

File: client/src/app/metadata-schema-form/custom/vf-tabs/vf-tabs.component.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import {AfterContentInit, Component, EventEmitter, Output} from '@angular/core';
+import {AfterContentInit, Component, EventEmitter, Output, Input, SimpleChanges, OnChanges} from '@angular/core';
```
[the remark this answers is not in the document]
Comment (jacobwindsor) [reply client/src/app/metadata-schema-form/custom/vf-tabs/vf-tabs.component.ts:1]: Yes

File: client/src/app/project-registration/project-registration-form/project-reg-layout.ts (hunk 1 above)
Comment (jacobwindsor) [reply client/src/app/project-registration/project-registration-form/project-reg-layout.ts:67]: Yes thanks

File: client/src/app/project-registration/project-registration-form/project-reg-layout.ts [hunk 5]
```diff
@@ -2,17 +2,26 @@ import {MetadataFormLayout} from '../../metadata-schema-form/models/metadata-for
...
     {
-      title: 'Project',
+      title: 'Publications',
+      key: 'project.content.publications',
+      items: [
+        'project.content.publications'
+      ]
+    },
+    {
+      title: 'Project information',
```
Comment (amnonkhen) [inline client/src/app/project-registration/project-registration-form/project-reg-layout.ts:19]: Do we use Title Case for titles?

File: client/src/app/metadata-schema-form/custom/vf-tabs/vf-tabs.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (amnonkhen) [reply client/src/app/metadata-schema-form/custom/vf-tabs/vf-tabs.component.ts:1]: I would create a ticket to move it out of this repo. I believe there should be a standalone repo for this, to make this integration easy for other projects as well.

File: client/src/app/project-registration/project-registration-form/project-reg-layout.ts (hunk 5 above)
Comment (amnonkhen) [reply client/src/app/project-registration/project-registration-form/project-reg-layout.ts:19]: ?

File: client/src/app/project-registration/project-registration-form/project-registration-form.component.html [hunk 6]
```diff
@@ -1,20 +1,33 @@
 <h1 class="vf-text vf-text-heading--2">{{title}}</h1>
-<p class="vf-text-body vf-text-body--3 vf-u-text-color--grey">
-  {{subtitle}}
-</p>
 <br/>
 <br/>
 
+<mat-tab-group>
+  <mat-tab label="1. Project">
```
Comment (amnonkhen) [inline client/src/app/project-registration/project-registration-form/project-registration-form.component.html:6]: the labels are duplicate. See project.compoment

**Review by amnonkhen — CHANGES_REQUESTED** [review]

**Review by amnonkhen — DISMISSED** [review]

File: client/src/app/metadata-schema-form/metadata-form/metadata-form.component.html [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <form *ngIf="done" class="vf-form" [formGroup]="metadataForm.formGroup" (submit)="onSubmit($event)">
   <ng-container *ngIf="visibleTabs.length > 1">
-    <app-vf-tabs *ngIf="useVf" (selectedIndexChange)="onSelectedIndexChange($event)">
+    <app-vf-tabs *ngIf="!!useVf" (selectedIndexChange)="onSelectedIndexChange($event)" [selectedIndex]="lookupTabIndex(selectedTabKey)">
```
Comment (aaclan-ebi) [inline client/src/app/metadata-schema-form/metadata-form/metadata-form.component.html:3]: just curious, why do `!!`?

File: client/src/app/shared/guards/wrangler-or-owner.guard.ts [hunk 8]
```diff
@@ -29,8 +32,8 @@ export class WranglerOrOwnerGuard implements CanActivate {
     }
 
     return ( getProject ?
-      this.isWranglerOrOwner(this.ingestService.getUserAccount(), getProject) :
-      this.isWrangler(this.ingestService.getUserAccount())
+      this.authService.isWranglerOrOwner(this.ingestService.getUserAccount(), getProject) :
+      this.authService.isWrangler(this.ingestService.getUserAccount())
```
Comment (aaclan-ebi) [inline client/src/app/shared/guards/wrangler-or-owner.guard.ts:36]: Suggestion: Could have the authService use the ingest service and cache the account there? Use the auth service to get the account. Then use account to know if it's a wrangler or owner of a project.

**Review by aaclan-ebi — APPROVED** [review]
Nice! I noticed you also fixed the missing _Upload_ button!

File: client/src/app/metadata-schema-form/metadata-form/metadata-form.component.html (hunk 7 above)
Comment (jacobwindsor) [reply client/src/app/metadata-schema-form/metadata-form/metadata-form.component.html:3]: Actually force of habit, not really necessary here. Ensures it's actually a boolean so e.g. `null` becomes `false`

File: client/src/app/shared/guards/wrangler-or-owner.guard.ts (hunk 8 above)
Comment (jacobwindsor) [reply client/src/app/shared/guards/wrangler-or-owner.guard.ts:36]: That's true. Can we do that in [#225](https://app.zenhub.com/workspaces/dcp-ingest-product-development-5f71ca62a3cb47326bdc1b5c/issues/ebi-ait/dcp-ingest-central/225)?

File: client/src/app/project-registration/project-registration-form/project-registration-form.component.html (hunk 6 above)
Comment (jacobwindsor) [reply client/src/app/project-registration/project-registration-form/project-registration-form.component.html:6]: We will resolve this in [#225](https://app.zenhub.com/workspaces/dcp-ingest-product-development-5f71ca62a3cb47326bdc1b5c/issues/ebi-ait/dcp-ingest-central/225)

File: client/src/app/shared/guards/wrangler-or-owner.guard.ts (hunk 8 above)
Comment (aaclan-ebi) [reply client/src/app/shared/guards/wrangler-or-owner.guard.ts:36]: Yup!

File: client/src/app/metadata-schema-form/metadata-form/metadata-form.component.html (hunk 7 above)
Comment (aaclan-ebi) [reply client/src/app/metadata-schema-form/metadata-form/metadata-form.component.html:3]: I see. Thanks!

File: client/src/app/project-registration/project-registration-form/project-reg-layout.ts (hunk 5 above)
Comment (jacobwindsor) [reply client/src/app/project-registration/project-registration-form/project-reg-layout.ts:19]: No

File: client/src/app/project/project.component.ts (hunk 3 above)
Comment (jacobwindsor) [reply client/src/app/project/project.component.ts:44]: Will fix in [#225](https://app.zenhub.com/workspaces/dcp-ingest-product-development-5f71ca62a3cb47326bdc1b5c/issues/ebi-ait/dcp-ingest-central/225)

File: client/src/app/metadata-schema-form/custom/vf-tabs/vf-tabs.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (jacobwindsor) [reply client/src/app/metadata-schema-form/custom/vf-tabs/vf-tabs.component.ts:1]: [done](https://app.zenhub.com/workspaces/dcp-ingest-product-development-5f71ca62a3cb47326bdc1b5c/issues/ebi-ait/dcp-ingest-central/242)

**Review by aaclan-ebi — APPROVED** [review]

**Review by yusra-haider — APPROVED** [review]

**Pull request merged**